Library/Homebrew/cmd/update.sh

```diff
@@ -336,23 +349,41 @@ EOS
         git fetch --force "${QUIET_ARGS[@]}" origin \
           "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" 2>/dev/null
       else
-        git fetch --force "${QUIET_ARGS[@]}" origin \
-          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" || \
-            odie "Fetching $DIR failed!"
+        if ! git fetch --force "${QUIET_ARGS[@]}" origin \
+          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH"
+        then
+          echo "Fetching $DIR failed!" >> "$HOMEBREW_UPDATE_FAILED_FILE"
```

Comment:
It's still racy, I think: the writes are non-atomic and non-locking AFAIK, so if there are two simultaneous write attempts, one of them might fail when the reads and writes get interleaved (e.g A opens for append & seeks to end; B opens for append & seeks to end; A writes at end+1 and closes; B writes at end+1 – using the old value of end – and closes). Odds of it happening are pretty low though.